src/tx/staking/verifyStake.ts

```diff
@@ -173,5 +172,34 @@ export function verifyUnstakeTx(
     reason = `This nominee node is not found!`
   }
 
+  if (!isStakeUnlocked(nominatorAccount, nomineeAccount, shardus)) {
+    success = false
+    reason = `The stake is not unlocked yet!`
+  }
+
   return { success, reason }
 }
+
+function isStakeUnlocked(
+  nominatorAccount: WrappedEVMAccount,
+  nomineeAccount: NodeAccount2,
+  shardus: Shardus
+): boolean {
+  const stakeLockTime = ShardeumFlags.stakeLockTime
+
+  // SLT from time of last staking or unstaking
+  if (shardus.shardusGetTime() - nominatorAccount.operatorAccountInfo.lastStakeTimestamp < stakeLockTime) {
+    return false
+  }
+
+  // SLT from time of last went active
+  // THIS MAY BE REDUNDANT as we check for active status in verifyUnstakeTx
+  if (shardus.shardusGetTime() - nomineeAccount.rewardStartTime * 1000 < stakeLockTime) {
+    return false
+  }
+
```

Comment:
I think there is one "potential hole" where a node can escape the stakeLockTime. This check would actually go in the if blocks above.   

I think we may need a isNodeGoingActiveByPubKey ...  this would be something that covers an of the states between selected to go active and when it is actually active.   Otherwise a node that had been in the standby list for a long time it could act up in this window and then unlock stake before there is a chance for the rewardStartTime to get set.  Either before it went active or after it was active but before initRewardTimes was processed

Reply:
an alternate way would be to look at a timer in the cycle record when the node was selected to go active.
This can be detected by using the syncingTimestamp.    you can see that this gets set in the cycle record when a node is selected to go active, so all nodes will know this setting for any node that has been selected to go active.     I think this will be more comprehensive and safe to future changes than my first suggestion